tox.ini

```diff
@@ -17,6 +17,7 @@ commands =
     flake8
 
 [flake8]
-ignore = H405,H404,H403,H401,H306,S101,N802,N803,N806,I202,I201
+# See https://gitlab.com/pycqa/flake8/-/issues/466 for extend-ignore vs ignore
+extend-ignore = H405,H404,H403,H401,H306,S101,N802,N803,N806,I202,I201
```

Comment:
All those "ignore" seems to indicate that we are not following standard python conventions ;)

Reply:
https://flake8.pycqa.org/en/latest/user/error-codes.html

Not sure what these are; I discovered that `extend-ignore` was having issues with W503/W504, where flake8 changed the behavior.

* H405, H404, H403, H401 are all about docstrings and spaces (I noticed some of these)
* S101 - Multi-line missing trailing comma
* N802, N803, N806 - function names lowercase, argument name lowercase, functiov in variable lowercase.
* I201, I202 - Newlines between import groups

Maybe we can clean these up in a separate PR before 1.0.0?

Reply:
FOR SURE we can change that in another PR!!! 
It was just a comment about this whole repo where there seems to have lot of ignored rules... 
Usually when we ignore rules it's because we don't like them or because we don't know the standard of the language.

Another thoughts here, personally I prefer `pylint` as linter (compare to `flake8`) since you can ignore rule by name instead of by "rule code" that are so hard (impossible) to remember.
For example, see this stackoverflow comment from 2014: https://stackoverflow.com/questions/4341746/how-do-i-disable-a-pylint-warning/23542817#23542817

Reply:
I totally agree on the code not feeling pythonic -- the existing code feels very Java-ish to me.

With that said, coming from go-land, I hate linters because it introduces human debug-fix loops where presumably the machine could do the formatting for me.

Reply:
I agree (these linting settings also predate me). I'd prefer if we just used https://pypi.org/project/black/, which includes an auto-formatter.